app/javascript/components/visualization/controls/GeneListSelector.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import Select from 'react-select'
+
+import { clusterSelectStyle } from 'lib/cluster-utils'
+
+// value to render in select menu if user has not selected a gene list
+const noneSelected = "None selected..."
+
+/** takes the server response and returns subsample options suitable for react-select */
+function getGeneListOptions(studyGeneLists) {
+  let geneListObjects = [{label: noneSelected, value: ''}]
+  geneListObjects = geneListObjects.concat(studyGeneLists.map(geneList => {
+    return { label: geneList, value: geneList }
+  }))
+  return geneListObjects
+}
```

Comment:
Consider:
```javascript
function getGeneListOptions(studyGeneLists) {
  const lAndV = x => ({label: x, value: x})
  return [{label: noneSelected, value: ''}].concat(studyGeneLists.map(lAndV))
}
```

Reply:
Ooh... fancy.  Functional programming to the rescue...

Reply:
75a566f